Sprint 6/step 1

Neverhood93 opened the pull request:
Добавлены библиотеки для тестирования.
Созданы функциональные и UI тесты.
Оформлен README.md.

Neverhood93 committed:
Установлены необходимые библиотеки для тестирования.

Neverhood93 committed:
Добавлена библиотека cypress.

Neverhood93 committed:
Установлен плагин eslint-plugin-cypress.
Настройка инфраструктуры для cypress.

Neverhood93 committed:
Добавлен тест для ingredientsSlice.

Neverhood93 committed:
Добавлен тест для orderSlice.

Neverhood93 committed:
Добавлен тест для authSlice.

Neverhood93 committed:
Добавлены тест для ingredientDetailsSlice и selectedIngredientsSlice.

Neverhood93 committed:
Добавлен тест для feedSlice.

Neverhood93 committed:
Добавлен тест для profileOrdersSlice.

Neverhood93 committed:
Тесты через Cypress:
Добавлена подмена коллекции ингредиентов через fixture.
Добавлены атрибуты "data-test-id" для поиска.
Добавлены тесты для открытия и закрытия модального окна разными способами.
Проверка детальной информации ингредиента.

Neverhood93 committed:
Реализован тест: перетаскивание ингредиента в конструктор.

Neverhood93 committed:
Изменен README.md.

mikebars1995 commented on cypress/e2e/constructor.test.spec.cy.js:
```diff
@@ -0,0 +1,57 @@
+  beforeEach(function () {
+    cy.intercept("GET", "ingredients", { fixture: "ingredients" });
+
+    cy.visit("http://localhost:3000");
```
**Нужно исправить**

`http://localhost:3000` обычно повторяется в коде тестов. Нужно вынести в константу `testUrl`, чтобы можно было одним движением изменить урл для тестов
А можно в документации посмотреть, как настроить быстро `baseUrl` для cypress https://docs.cypress.io/api/commands/visit#Visit-is-automatically-prefixed-with-baseUrl

mikebars1995 commented on cypress/e2e/constructor.test.spec.cy.js:
```diff
@@ -0,0 +1,57 @@
+
+  it("should open modal with ingredient detail and close with close button", () => {
+    cy.get('[data-test-id="643d69a5c3f7b9001cfa093c"]').click({ force: true });
+    cy.get('[data-test-id="modal_box"]').should("exist");
```
**Нужно исправить**

Если **строка-селектор** повторяется более 2х раз в коде, значит, этот селектор нужно выносить в **константу**, чтобы можно было одним движением изменить его и не искать по всему коду дублирования этого селектора.

mikebars1995 reviewed (commented):
Здравствуйте. (**Нужно развернуть общий комментарий ↓**)

Посмотрите на **гитхабе** все комментарии к коду (нужно прокрутить вниз страницу там)

Работа проделана огромная:


* Все юнит-тесты (59 шт) проходят успешно
* Отлично, что знаете про `alias` (переменные с `@`)
* Код тестов аккуратный и понятный


но есть некоторые недочеты:

* `http://localhost:3000` обычно повторяется в коде тестов. Нужно вынести в константу `testUrl`, чтобы можно было одним движением изменить урл для тестов
* Если **строка-селектор** повторяется более 2х раз в коде, значит, этот селектор нужно выносить в **константу**, чтобы можно было одним движением изменить его и не искать по всему коду дублирования этого селектора. 
* Нужно добавить скрипт для запуска cypress-тестов

Neverhood93 committed:
Исправление замечаний:
Добавлена команда для запуска cypress тестов.
Изменен cypress.config.ts.
Добавлены константы для строк-селекторов selectors.js для тестов cypress.

mikebars1995 reviewed (commented):
Поздравляю! Ваша работа принята.

Вы отлично потрудились.

Удачного дальнейшего обучения.
